### PR TITLE
fix: Server crashes when receiving an array of `Parse.Pointer` in the request body

### LINKED ISSUE #8783: Parse server crash  - Cannot read property 'encodeParseObjectInCloudFunction' of undefined
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

Parse server crash  - Cannot read property 'encodeParseObjectInCloudFunction' of undefined

### Steps to reproduce

I cannot send you my network body its too complex to explain, but its easy to see in the code
### Actual Outcome

Parse server crash  - Cannot read property 'encodeParseObjectInCloudFunction' of undefined
### Expected Outcome

Not to crash 
### Environment

Prase server 6.3.1

FunctionsRouter.js  - [line 15](https://github.com/parse-community/parse-server/blob/fe02d3e8aa875f362377cc877a65831b010b00a6/src/Routers/FunctionsRouter.js#L15) does not pass the config object to the parseObject function
That will crash FunctionsRouter.js  - [line 21](https://github.com/parse-community/parse-server/blob/fe02d3e8aa875f362377cc877a65831b010b00a6/src/Routers/FunctionsRouter.js#L21) config.encodeParseObjectInCloudFunction because the config is undefined in this case

Please pass config object in line 15 and maybe add config?.encodeParseObjectInCloudFunction 

Thanks

### PR DESCRIPTION
**Pull request by zivchen**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

Closes: https://github.com/parse-community/parse-server/issues/8783

## Tasks

- [x] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)
- [ ] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [ ] Add new Parse Error codes to Parse JS SDK

### CODE REVIEW

**Comment by mtrezza** [comment]
Could you please add a test that demonstrates the server crash?

**Review by sadortun — APPROVED** [review]

**Comment by sadortun** [comment]
@mtrezza I'm probably 100% looking at the wrong place, but where are the tests for `parseObject()` ?

**Comment by mtrezza** [comment]
Did you look at the blame line to see whether there were test in the PR?

**Comment by sadortun** [comment]
> Did you look at the blame line to see whether there were test in the PR?

Yes, and i don't really see test-cases covering the `parseObject()`   in your original commit where the original change were made. But i might be looking at the wrong place, or ` parseObject()` is tested indirectly.

https://github.com/parse-community/parse-server/commit/e212eb51954bfb5c49a59689fd785740489712fc#diff-8f2f0b3d469b333004223792d072d82895edc96039eae3dba67a4eb4e37ae66dR12

The missing test could simply add `[]`  to test  the `.map()`

Ex:  ` parseObject( [item]  , config);`

**Comment by rdhelms** [comment]
Would love to see this fixed - we just tried to update and ran into this error

**Comment by pocketcolin** [comment]
@sadortun I'm not a Parse maintainer but I think I figured out where this test should live and created a PR to the forked repo to add one that verifies this fix: https://github.com/zivchen/parse-server/pull/2

**Comment by pocketcolin** [comment]
I just realized that the original creator of this PR hasn't replied since October 22 so I decided to go ahead and create a new PR with the test and fix so we can get it merged sooner. If the original creator comes back and wants to use their PR instead, that's also totally fine!

https://github.com/parse-community/parse-server/pull/8832

**Comment by zivchen** [comment]
Hi
I merged @pocketcolin test spec into the PR
Thanks

**Comment by pocketcolin** [comment]
Awesome! Should we also create a PR for a v6 hotfix? This PR just fixes the issue in the v7 alpha and who knows when v7 will be ready for prod.

**Comment by mtrezza** [comment]
@zivchen Could you please edit the PR title so that it makes a meaningful change log entry where developers understand what issue this fixes? Can you confirm that this PR fixes the issue you faced?

**Comment by mtrezza** [comment]
Could you rephrase the PR title so that it's not describing the source code but the use case which causes the crash? It seems this only happens when using a Pointer, right?

**Comment by gtzinos** [comment]
Just dropping a line to say we're eagerly eyeing this pull request, like kids at a candy store window. 😊

**Review by vasileios-s — APPROVED** [review]

**Review by JohnStamtsidis — APPROVED** [review]

**Review by vaskouk — APPROVED** [review]

File: spec/ParseAPI.spec.js
Comment (mtrezza) [inline spec/ParseAPI.spec.js]: Please rewrite using await; we are avoiding promise chains.

File: spec/ParseAPI.spec.js
Comment (mtrezza) [inline spec/ParseAPI.spec.js]: What is this query comment for, can it be removed?

File: spec/ParseAPI.spec.js
Comment (mtrezza) [inline spec/ParseAPI.spec.js]: Is this necessary, given that the query string is commented out? It looks like you were testing something additional (query param overwrites body param) that is beyond the test's purpose.

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: spec/ParseAPI.spec.js
Comment (mtrezza) [inline spec/ParseAPI.spec.js]: Is this actually a valid pointer object? It seems to be missing the classname and objectId?

Do we need `foo: bar` and `other: 1` in this test?

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by mtrezza** [comment]
We'll move Parse Server 6 to a LTS branch so a PR can be opened for it as well.

**Comment by mtrezza** [comment]
@zivchen Just a friendly ping regarding the feedback; it would be great to get this PR merged.

File: spec/ParseAPI.spec.js
Comment (mtrezza) [inline spec/ParseAPI.spec.js]:
```suggestion
  it('test cloud function query parameters with array of pointers', async () => {
    Parse.Cloud.define('echoParams', req => {
      return req.params;
    });
    const headers = {
      'Content-Type': 'application/json',
      'X-Parse-Application-Id': 'test',
      'X-Parse-Javascript-Key': 'test',
    };
    const response = await request({
      method: 'POST',
      headers: headers,
      url: 'http://localhost:8378/1/functions/echoParams',
      body: '{"arr": [{ "__type": "Pointer", "className": "PointerTest" }]}',
    });
    const res = response.data.result;
    expect(res.arr.length).toEqual(1);
  });
```

Comment (mtrezza) [reply spec/ParseAPI.spec.js]: remove `done`, we're not using this anymore

**Review by mtrezza — APPROVED** [review]
Looks good, waiting for CI to pass...

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0-alpha.9](https://github.com/parse-community/parse-server/releases/tag/7.0.0-alpha.9)

**Comment by jaysonng** [comment]
> Awesome! Should we also create a PR for a v6 hotfix? This PR just fixes the issue in the v7 alpha and who knows when v7 will be ready for prod.

yes pls. I need it for v6 and for the life of me can't figure out the proper procedure to do a PR for v6.

**Comment by szymon-wrobel-pronos** [comment]
@mtrezza Is there a timeline for carrying over this bugfix to the v6 version?

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/7.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0](https://github.com/parse-community/parse-server/releases/tag/7.0.0)

**Pull request merged**
